runtime/doc/dev_style.txt

```diff
@@ -0,0 +1,1157 @@
+functions. >
+
+    <HEADER>
+
+    <PUBLIC FUNCTION DEFINITIONS>
+
+    <STATIC FUNCTION DEFINITIONS>
+
+
+Integration with declarations generator ~
+
+Every C file must contain #include of the generated header file, guarded by
+#ifdef INCLUDE_GENERATED_DECLARATIONS.
+
+Include must go after other #includes and typedefs in .c files and after
+everything else in header files. It is allowed to omit #include in a .c file
+if .c file does not contain any static functions.
+
+Included file name consists of the .c file name without extension, preceded by
+the directory name relative to src/nvim. Name of the file containing static
+functions declarations ends with `.c.generated.h`, `*.h.generated.h` files
+contain only non-static function declarations. >
+
+    // src/nvim/foo.c file
+    #include <stddef.h>
+
+    typedef int FooType;
+
+    #ifdef INCLUDE_GENERATED_DECLARATIONS
+    # include "foo.c.generated.h"
+    #endif
+
+    …
+
+
+    // src/nvim/foo.h file
+    #ifndef NVIM_FOO_H
+    #define NVIM_FOO_H
+
+    …
+
+    #ifdef INCLUDE_GENERATED_DECLARATIONS
+    # include "foo.h.generated.h"
+    #endif
+    #endif  // NVIM_FOO_H
+
+
+64-bit Portability ~
+
+Code should be 64-bit and 32-bit friendly. Bear in mind problems of printing,
+comparisons, and structure alignment.
+
+- Remember that `sizeof(void *)` != `sizeof(int)`. Use `intptr_t` if you want
+  a pointer-sized integer.
+
+- You may need to be careful with structure alignments, particularly for
+  structures being stored on disk. Any class/structure with a
+  `int64_t`/`uint64_t` member will by default end up being 8-byte aligned on a
+  64-bit system. If you have such structures being shared on disk between
+  32-bit and 64-bit code, you will need to ensure that they are packed the
+  same on both architectures. Most compilers offer a way to alter structure
+  alignment. For gcc, you can use `__attribute__((packed))`. MSVC offers
+  `#pragma pack()` and `__declspec(align())`.
+
+- Use the `LL` or `ULL` suffixes as needed to create 64-bit constants. For
+  example: >
+
+    int64_t my_value = 0x123456789LL;
+    uint64_t my_mask = 3ULL << 48;
+
+
+sizeof ~
+
+Prefer `sizeof(varname)` to `sizeof(type)`.
+
+Use `sizeof(varname)` when you take the size of a particular variable.
+`sizeof(varname)` will update appropriately if someone changes the variable
+type either now or later. You may use `sizeof(type)` for code unrelated to any
+particular variable, such as code that manages an external or internal data
+format where a variable of an appropriate C type is not convenient. >
+
+    Struct data;
+    memset(&data, 0, sizeof(data));
+
+    memset(&data, 0, sizeof(Struct));
+
+    if (raw_size < sizeof(int)) {
+      fprintf(stderr, "compressed record not big enough for count: %ju", raw_size);
+      return false;
+    }
+
+
+==============================================================================
+Naming								*style-naming*
```

Comment:
all of the tags in this file should be prefixed with `dev-`. They are only relevant to core devs.


```suggestion
Naming								*dev-style-naming*
```